### PR TITLE
alru: add parameter `max_dirty_ratio'

### PR DESCRIPTION
**Pull request by live4thee**
With a high dirty ratio and occupancy, OCF might unable to map cache lines for new requests, thus pass-through the I/O to core devices.  IOPS will drop afterwards.  Existing `alru' policy gives user the chance to control the stale buffer time, activity threshold etc.  They can affect the dirty ratio of the cache device, but in an empirical manner, more or less.  Introducing `max_dirty_ratio' can make it explicit.

At first glance, it might be better to implement a dedicated cleaner policy directly targeting dirty ratio goal, so that the `alru' parameters remains orthogonal.  But one the other hand, we still need to flush dirty cache lines periodically, instead of just keeping a watermark of dirty ratio. It indicates that existing `alru' parameters are still required if we develop a new policy, and it seems reasonable to make it a parameter.

To sum up, this patch does the following:
- added a 'max_dirty_ratio' parameter with default value 100;
- with default value 100, `alru' cleaner is identical to what is was;
- with value N less than 100, the cleaner (when waken up) will active brought dirty ratio to N, regardless of staleness time.

I  have tested this patch with a modified Open-CAS and it works. If any comments, please feel free.

### CODE REVIEW

**Review by robertbaldyga — APPROVED** [review]
@live4thee Thanks for the contribution! The change looks good to me.
Please add your company (or your personal) copyright line in headers of files you changed.

**Review by kmajzero — APPROVED** [review]

**Review by kmajzero — APPROVED** [review]
Please update the copyright header at the beginning of alru.h with a correct date.
